codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -153,11 +174,15 @@ private static ByteBuf encodeConnectMessage(
         buf.writeByte(variableHeader.version());
         buf.writeByte(getConnVariableHeaderFlag(variableHeader));
         buf.writeShort(variableHeader.keepAliveTimeSeconds());
+        buf.writeBytes(propertiesBuf);
+        propertiesBuf.release();
```

Comment:
I wonder if we should better do the release in a finally block. Just in case if something goes wrong... otherwise we will leak memory